packages/sdk/src/reads/getCurveBestPrice.ts

```diff
@@ -0,0 +1,84 @@
+import { WETH } from "../../tests/constants.js";
+import { ETH_ADDRESS } from "../constants/misc.js";
+import { invariant } from "../utils/assertions.js";
+import { type ReadContractParameters, readContractParameters } from "../utils/viem.js";
+import { type Address, type PublicClient, parseAbi, zeroAddress } from "viem";
+
+const abi = {
+  name: "get_address",
+  outputs: [{ type: "address", name: "" }],
+  inputs: [{ type: "uint256", name: "_id" }],
+  stateMutability: "view",
+  type: "function",
+  gas: 1308,
+} as const;
+
+type Asset = {
+  address: Address;
+  decimals: number;
+  symbol: string;
+};
+
+const curveRegistry = "0x0000000022d53366457f9d5e68ec105046fc4383" as const;
+const swapId = 2n; // id won't change, for swaps it will be always the same id in the registry
+
+export async function getCurveBestPrice(
+  client: PublicClient,
+  args: ReadContractParameters<{
+    incoming: Asset;
+    outgoing: Asset;
+    quantity: bigint;
+  }>,
+) {
+  invariant(args.incoming.address !== args.outgoing.address, "Incoming and outgoing asset are identical");
```

Comment:
it is for sure not invariant, but I don't think we should do validation here at all, we don't do that in other places